easybuild/easyblocks/c/cp2k.py

```diff
@@ -228,12 +228,30 @@ def configure_step(self):
             else:
                 options['DFLAGS'] += ' -D__ELPA3'
 
-        # CUDA
+        # CUDA support
+        # see https://github.com/cp2k/cp2k/blob/master/INSTALL.md#2j-cuda-optional-improved-performance-on-gpu-systems
         cuda = get_software_root('CUDA')
         if cuda:
-            options['DFLAGS'] += ' -D__ACC -D__DBCSR_ACC'
-            options['LIBS'] += ' -lcudart -lcublas -lcufft -lrt'
-            options['NVCC'] = ' nvcc'
+            # check GPU architecture(s) to build for
+            cuda_cc_sm_vals = self.cfg.template_values.get('cuda_sm_comma_sep')
+            if cuda_cc_sm_vals:
+                gpu_targets = '--gpu-code=%s' % cuda_cc_sm_vals
+            else:
+                gpu_targets = ''
+
+            options['DFLAGS'] += ' -D__ACC -D__DBCSR_ACC -D__PW_CUDA -D__GRID_CUDA'
+            options['LIBS'] += ' -lcudart -lnvrtc -lcuda -lcublas -lcufft -lrt'
+            options['NVCC'] = 'nvcc'
+            options['NVFLAGS'] = ' '.join([
+                options['DFLAGS'],
+                '-O3',
+                '--std=c++11',
+                gpu_targets,
+                # control host compilers + options
+                '-ccbin="%s"' % os.getenv('CXX'),
+                '-Xcompiler="%s"' % os.getenv('CXXFLAGS'),
+            ])
+            options['GPUVER'] = 'V100'
```

Comment:
This is obviously hardcoded, should we auto-detect what the correct value should be for this?

Not sure what the significance of this is, actually...

Reply:
Is this the point where you said in the call that only a single GPU arch is supported?

I just searched the source of CP2K and didn't find any place where this is actually used: https://github.com/cp2k/cp2k/search?q=GPUVER
All uses are either in the autotools build chain, which we don't use (I'd guess we could?) So I guess we don't need it at all. Or did I miss anything?

Reply:
No, this is really required, I tried without it. It's used by the DBCSR component.

I have an updated version of this, which I apparently haven't pushed yet, doing that right now (see 68e15c56bcb13f685695a1315f27489ee1acdf3c)